src/client/Microsoft.Identity.Client/WwwAuthenticateParameters.cs

```diff
@@ -0,0 +1,334 @@
+                httpResponseHeaders,
+                scheme);
+
+            try
+            {
+                // read the header and checks if it contains an error with insufficient_claims value.
+                if (null != parameters.Error && "insufficient_claims" == parameters.Error)
+                {
+                    if (null != parameters.Claims)
+                    {
+                        return parameters.Claims;
+                    }
+                }
+            }
+            catch (Exception ex)
+            {
+                throw ex;
+            }
+
+            return null;
+        }
+
+        internal static WwwAuthenticateParameters CreateWwwAuthenticateParameters(IDictionary<string, string> values)
+        {
+            WwwAuthenticateParameters wwwAuthenticateParameters = new WwwAuthenticateParameters();
+            wwwAuthenticateParameters.RawParameters = values;
+            string value;
+
+            if (values.TryGetValue("authorization_uri", out value))
+            {
+                wwwAuthenticateParameters.Authority = value.Replace("/oauth2/authorize", string.Empty);
+            }
+
+            if (string.IsNullOrEmpty(wwwAuthenticateParameters.Authority))
+            {
+                if (values.TryGetValue("authorization", out value))
+                {
+                    wwwAuthenticateParameters.Authority = value.Replace("/oauth2/authorize", string.Empty);
+                }
+            }
+
+            if (string.IsNullOrEmpty(wwwAuthenticateParameters.Authority))
+            {
+                if (values.TryGetValue("authority", out value))
+                {
+                    wwwAuthenticateParameters.Authority = value.TrimEnd('/');
+                }
+            }
+
+            if (values.TryGetValue("resource_id", out value))
+            {
+                wwwAuthenticateParameters.Resource = value;
+            }
+
+            if (string.IsNullOrEmpty(wwwAuthenticateParameters.Resource))
+            {
+                if (values.TryGetValue("resource", out value))
+                {
+                    wwwAuthenticateParameters.Resource = value;
+                }
+            }
+
+            if (string.IsNullOrEmpty(wwwAuthenticateParameters.Resource))
+            {
+                if (values.TryGetValue("client_id", out value))
+                {
+                    wwwAuthenticateParameters.Resource = value;
+                }
+            }
+
+            if (values.TryGetValue("claims", out value))
+            {
+                wwwAuthenticateParameters.Claims = GetJsonFragment(value);
+            }
+
+            if (values.TryGetValue("error", out value))
+            {
+                wwwAuthenticateParameters.Error = value;
+            }
+
+            return wwwAuthenticateParameters;
+        }
+
+        internal static List<string> SplitWithQuotes(string input, char delimiter)
```

Comment:
>internal static List<string> SplitWithQuotes(string input, char delimiter) [](http://example.com/codeflow?start=8&length=74)

wondering if we can optimize string usage with span's here

Reply:
i took this from a mise sample for their www-auth processing. will look into spans

Reply:
i think i'm going to leave it as-is for now.